test/unit/content-info.test.js

```diff
@@ -0,0 +1,629 @@
+            },
+            {
+                key: 'Content-Type',
+                value: 'application/json'
+            }
+        ], stream: Buffer.from('a test json').toJSON()
+        });
+        expect(contentInfo.contentInfo(response)).to
+            .eql({
+                charset: 'utf8',
+                fileExtension: 'json',
+                mimeFormat: 'json',
+                mimeType: 'text',
+                fileName: '响应.json'
+            });
+    });
+
+    it('Should return file name from content-disposition header with encoded type iso-8859-1', function () {
+        var response = new Response({ header: [
+            {
+                key: 'Content-Disposition',
+                value: 'attachment;filename*=iso-8859-1\'\'myResponse.json'
+            },
+            {
+                key: 'Content-Type',
+                value: 'application/json'
+            }
+        ], stream: Buffer.from('a test json').toJSON()
+        });
+        expect(contentInfo.contentInfo(response)).to
+            .eql({
+                charset: 'utf8',
+                fileExtension: 'json',
+                mimeFormat: 'json',
+                mimeType: 'text',
+                fileName: 'myResponse.json'
+            });
+    });
+
+    it('filename* parameter is preferred than filename in content-disposition header', function () {
+        var response = new Response({ header: [
+            {
+                key: 'Content-Disposition',
+                value: 'attachment;filename*=utf-8\'\'%E5%93%8D%E5%BA%94.json; filename = response.json'
+            },
+            {
+                key: 'Content-Type',
+                value: 'application/json'
+            }
+        ], stream: Buffer.from('a test json').toJSON()
+        });
+        expect(contentInfo.contentInfo(response)).to
+            .eql({
+                charset: 'utf8',
+                fileExtension: 'json',
+                mimeFormat: 'json',
+                mimeType: 'text',
+                fileName: '响应.json'
+            });
+    });
+
+    it('Should return file name from content-disposition header with type form-data and file name', function () {
+        var response = new Response({ header: [
+            {
+                key: 'Content-Disposition',
+                value: 'form-data; filename="testResponse.json"'
+            },
+            {
+                key: 'Content-Type',
+                value: 'application/json'
+            }
+        ], stream: Buffer.from('a test json').toJSON()
+        });
+        expect(contentInfo.contentInfo(response)).to
+            .eql({
+                charset: 'utf8',
+                fileExtension: 'json',
+                mimeFormat: 'json',
+                mimeType: 'text',
+                fileName: 'testResponse.json'
+            });
+    });
+
+    describe('edge cases', function () {
+        it('Should take default filename if content-disposition header value is empty', function () {
```

Comment:
Should we also not have test cases where one or both of the headers are missing?